pulp_container/tests/functional/api/test_sync.py

```diff
@@ -1,231 +1,147 @@
 
-        1. Create a repository, and a remote.
-        2. Assert that repository version is None.
-        3. Sync the remote.
-        4. Assert that repository version is not None.
-        5. Sync the remote one more time.
-        6. Assert that repository version is the same from the previous one.
-        """
-        repository_api = RepositoriesContainerApi(self.client_api)
-        repository = repository_api.create(ContainerContainerRepository(**gen_repo()))
-        self.addCleanup(repository_api.delete, repository.pulp_href)
+    sync(container_remote, container_repo)
+    repository = container_repository_api.read(container_repo.pulp_href)
 
-        remote_api = RemotesContainerApi(self.client_api)
-        remote = remote_api.create(gen_container_remote())
-        self.addCleanup(remote_api.delete, remote.pulp_href)
+    assert repository.latest_version_href
 
-        self.assertEqual(repository.latest_version_href, f"{repository.pulp_href}versions/0/")
-        repository_sync_data = ContainerRepositorySyncURL(remote=remote.pulp_href)
+    latest_version_href = repository.latest_version_href
+    sync(container_remote, container_repo)
+    repository = container_repository_api.read(repository.pulp_href)
 
-        # Sync the repository.
-        sync_response = repository_api.sync(repository.pulp_href, repository_sync_data)
-        monitor_task(sync_response.task)
-        repository = repository_api.read(repository.pulp_href)
-        self.assertIsNotNone(repository.latest_version_href)
+    assert repository.latest_version_href == latest_version_href
 
-        # Sync the repository again.
-        latest_version_href = repository.latest_version_href
-        sync_response = repository_api.sync(repository.pulp_href, repository_sync_data)
-        monitor_task(sync_response.task)
-        repository = repository_api.read(repository.pulp_href)
-        self.assertEqual(latest_version_href, repository.latest_version_href)
 
+@pytest.mark.parallel
+def test_sync_invalid_url(container_repo, container_remote_api, sync, gen_object_with_cleanup):
+    remote_data = gen_container_remote(url="http://i-am-an-invalid-url.com/invalid/")
+    container_remote = gen_object_with_cleanup(
+        container_remote_api, ContainerContainerRemote(**remote_data)
+    )
 
-class SyncInvalidURLTestCase(PulpTestCase):
-    """Sync a repository with an invalid url on the Remote."""
+    with pytest.raises(PulpTaskError) as ctx:
+        sync(container_remote, container_repo)
 
-    def test_all(self):
-        """
-        Sync a repository using a Remote url that does not exist.
+    assert "[Name or service not known]" in ctx.value.task.error["description"]
 
-        Test that we get a task failure.
 
-        """
-        client_api = gen_container_client()
+@pytest.mark.parallel
+def test_sync_idempotency(
+    container_repo,
+    container_remote,
+    container_tag_api,
+    sync,
+):
```

Comment:
Can you preserve the docstring of tests (without the plan.io link)?
It helps to understand what the test is meant to do. Also does this test assert anything new compared to `test_basic_sync`?